packages/dds/tree/src/test/simple-tree/toMapTree.spec.ts

```diff
@@ -7,23 +7,47 @@ import { strict as assert } from "assert";
 
 import { MockHandle, validateAssertionError } from "@fluidframework/test-runtime-utils/internal";
 
-import { EmptyKey, type FieldKey, type MapTree } from "../../core/index.js";
+import {
+	EmptyKey,
+	LeafNodeStoredSchema,
+	ValueSchema,
+	type FieldKey,
+	type FieldKindData,
+	type FieldKindIdentifier,
+	type MapTree,
+	type SchemaAndPolicy,
+	type TreeNodeSchemaIdentifier,
+	type TreeNodeStoredSchema,
+} from "../../core/index.js";
 import { leaf } from "../../domains/index.js";
 import { SchemaFactory } from "../../simple-tree/index.js";
 // eslint-disable-next-line import/no-internal-modules
 import type { InsertableContent } from "../../simple-tree/proxies.js";
-// eslint-disable-next-line import/no-internal-modules
-import { ImplicitAllowedTypes, normalizeAllowedTypes } from "../../simple-tree/schemaTypes.js";
-// eslint-disable-next-line import/no-internal-modules
-import { nodeDataToMapTree as nodeDataToMapTreeBase } from "../../simple-tree/toMapTree.js";
+import {
+	FieldKind,
+	createFieldSchema,
+	ImplicitAllowedTypes,
+	normalizeAllowedTypes,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../simple-tree/schemaTypes.js";
+import {
+	cursorFromFieldData,
+	cursorFromNodeData,
+	nodeDataToMapTree as nodeDataToMapTreeBase,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../simple-tree/toMapTree.js";
 import { brand } from "../../util/index.js";
 
 /**
  * Wrapper around {@link nodeDataToMapTreeBase} which handles the normalization of {@link ImplicitAllowedTypes} as a
  * convenience.
  */
-function nodeDataToMapTree(tree: InsertableContent, allowedTypes: ImplicitAllowedTypes): MapTree {
-	return nodeDataToMapTreeBase(tree, normalizeAllowedTypes(allowedTypes));
+function nodeDataToMapTree(
+	tree: InsertableContent,
+	allowedTypes: ImplicitAllowedTypes,
+	schemaAndPolicy: SchemaAndPolicy | undefined = undefined,
```

Comment:
(Not a leading question - I'm curious) The reason this object is passed around is to inform policy around schema validation. We don't really need the schema portion, except as it is needed for validation. I wonder if it might make sense to call the parameter (here and elsewhere) `schemaValidationPolicy` or something.

Reply:
Yeah, I had a few debates with myself around naming this. I think your suggestion makes sense. @noencke what's your take?

Reply:
I went ahead with the rename, I think it makes sense. We can easily change it if we want.